common/math/geometry/src/solver/polynomial_solver.cpp

```diff
@@ -38,9 +38,8 @@ double PolynomialSolver::quadraticFunction(double a, double b, double c, double
 std::vector<double> PolynomialSolver::solveLinearEquation(
   double a, double b, double min_value, double max_value) const
 {
-  constexpr double e = std::numeric_limits<double>::epsilon();
-  if (std::fabs(a) < e) {
-    if (std::fabs(b) < e) {
+  if (std::abs(a) < tolerance) {
```

Comment:
This conditional expression is, after all, equivalent to
```
  if (std::abs(a) < tolerance) {
    return 0;
  }
```
Either you don't understand the code you're writing, or there's likely a bug lurking here.

Reply:
I checked the formulation of the logic and I found that old version of the code returns the wrong value when the a and b value is small.
So, I changed the specification of the function.

Reply:
Before:
:o:  `ax + b = 0 (a=0, b≠0)` => return {} (it means, there is no solution.)
:x:  `ax + b = 0 (a=0, b=0)` and `(min_value <= 0 && 0 <= max_value) == false`=> return {} (it means, there is no solution. But actual, any rational number between min_value ~ max_value is a solution. )
:x: `ax + b = 0 (a=0, b=0)` and `(min_value <= 0 && 0 <= max_value) == true` => return {0} (it means, 0 is a solution.  But actual, any rational number between min_value ~ max_value is a solution. )

After:
:o: `ax + b = 0 (a=0, b≠0)` => return {} (it means, there is no solution.)
:o: `ax + b = 0 (a=0, b=0)` => throw SimulationError (Any rational number between min_value ~ max_value is a solution, so infinite numbers of solutions are exist.)